test/staticClient/headers.test.ts

```diff
@@ -254,5 +296,60 @@ describe("Base Client headers", () => {
       });
       expect(nock.isDone()).to.be.true;
     });
+
+    it("overrides duplicate headers from client", async () => {
+      const client = new BaseClient({
```

Comment:
how about a test with

- client with default `user-agent` header with value as "client-header"
- a request with `user-agent` header with value as "request-header"
- verify nock with headers `user-agent` with value as "client-header request-header"

Reply:
With a header specified on both the client and the request, the behavior is to override, rather than merge (i.e. `request-header` rather than `client-header request header`). This test block ("overrides duplicate headers from client") tests that behavior.

There is no special logic for the user agent in core, it's just an extra value set by the SDK.

Reply:
IMHO, the behavior must be a `merge`. is there a way that merge can be accomplished with the current changes?

Reply:
A merge can be done, but it would be a breaking change. The behavior of the SDK has always been to override, rather than merge.

Reply:
i think we discussed about merging the headers in our team discussion. well, I am ok with restoring the overriding behavior and creating a ticket to address merge separately..

Reply:
There are different kinds of merging happening, which probably caused some confusion.
* Different-cased headers specified on each source are merged with headers _from that source_.
* Headers specified on the request _override_ those specified on the client.
* In the generated SDK, our custom user agent is treated as any other header. It is _merged_ with other user agent headers specified on the request, and then it _overrides_ any specified on the client.